phpunit/webfonts-api/class-wp-webfonts-schema-validator-test.php

```diff
@@ -0,0 +1,476 @@
+<?php
+
+/**
+ * @group  webfonts
+ * @covers WP_Webfonts_Schema_Validator
+ */
+class WP_Webfonts_Schema_Validator_Test extends WP_UnitTestCase {
+	private static $validator;
+
+	public static function setUpBeforeClass() {
+		require_once dirname( dirname( __DIR__ ) ) . '/lib/webfonts-api/class-wp-webfonts-schema-validator.php';
+
+		self::$validator = new WP_Webfonts_Schema_Validator();
+	}
+
+	/**
+	 * @covers WP_Webfonts_Schema_Validator::is_valid_schema
+	 *
+	 * @dataProvider data_is_valid_schema_with_valid
+	 *
+	 * @param array $webfont Webfont input.
+	 */
+	public function test_is_valid_schema_with_valid( array $webfont ) {
+		$this->assertTrue( self::$validator->is_valid_schema( $webfont ) );
+	}
+
+	/**
+	 * Data Provider.
+	 *
+	 * @return array
+	 */
+	public function data_is_valid_schema_with_valid() {
+		return array(
+			'basic schema' => array(
+				array(
+					'provider'    => 'google',
+					'font-family' => 'Open Sans',
+					'font-style'  => 'normal',
+					'font-weight' => '400',
+				),
+			),
+		);
+	}
+
+	/**
+	 * @covers  WP_Webfonts_Schema_Validator::is_valid_schema
+	 *
+	 * @dataProvider data_is_valid_schema_with_invalid
+	 *
+	 * @param array  $webfont          Webfont input.
+	 * @param string $expected_message Expected notice message.
+	 */
+	public function test_is_valid_schema_with_invalid( array $webfont, $expected_message ) { // phpcs:ignore VariableAnalysis
+
+		/*
+		$this->expectNotice();
+		$this->expectNoticeMessage( $expected_message );
+		*/
```

Comment:
These are from PHPUnit 9.5. In Core, it uses PHPUnit Polyfills to backwards shim for cross-version compatibility. But Gutenberg's repo does not yet have this. Rather, it runs on one PHP version with PHPUnit version 7.5.20. So this code needs to be changed for Gutenberg (for now).

```suggestion
		$this->expectException( Notice::class );
		$this->expectException( $expected_message );
```

and at the top of the file after the file's docbock add:

```
use PHPUnit\Framework\Error\Notice as Notice
```